src/main/java/com/basho/riak/client/core/RiakNode.java

```diff
@@ -877,6 +901,21 @@ public void onException(Channel channel, final Throwable t)
         }
     }
 
+    private void checkNetworkAddressCacheSettings()
+    {
+        final String property = Security.getProperty("networkaddress.cache.ttl");
+
+        int cacheTTL = property == null ? -1 : Integer.parseInt(property);
+
+        if (cacheTTL == -1)
+        {
+            logger.warn(
+                    "networkaddress.cache.ttl is currently set to cache DNS lookups forever. If you use domain names " +
+                    "for the Riak server connection, and an IP address changes (riak host or a load balancer), " +
+                    "this will block the client from creating connections with the new IP addresses.");
```

Comment:
If the property is null then there is no security manager in use, and the default is implementation dependent (30s in JRE7). They could still set it to -1 themselves (or by a zealous admin).  Going to check for both ways.